contributors/devel/sig-instrumentation/metric-stability.md

```diff
@@ -0,0 +1,101 @@
+
+An example of a stable metric follows:
+
+```go
+var deprecatedMetricDefinition = kubemetrics.CounterOpts{
+    Name: "some_deprecated_metric",
+    Help: "some description",
+    StabilityLevel: kubemetrics.STABLE, // this is also a custom metadata field
+    DeprecatedVersion: "1.15", // this is a custom metadata field
+}
+```
+
+By *not change*, we mean three things:
+
+1. the metric itself will not be deleted ([or renamed](#metric-renaming))
+2. the type of metric will not be modified
+3. no labels can be added **or** removed from this metric
+
+From an ingestion point of view, it is backwards-compatible to add or remove possible __values__ for labels which already do exist (but __not__ labels themselves). Therefore, adding or removing __values__ from an existing label is permissible. Stable metrics can also be marked as __deprecated__ for a future Kubernetes version, since this is a metadata field and does not actually change the metric itself.
+
+**Removing or adding labels from stable metrics is not permissible.** In order to add/remove a label to an existing stable metric, one would have to introduce a new metric and deprecate the stable one; otherwise this would violate compatibility agreements.
+
+## API Review
+
+Graduating a metric to a stable state is a contractual API agreement, as such, it would be desirable to require an api-review (to sig-instrumentation) for graduating or deprecating a metric (in line with current Kubernetes [api-review processes](https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md)).
+
+We use a verification script to flag stable metric changes for review by SIG Instrumentation approvers.
+
+## Metric Renaming
+
+Metric renaming is be tantamount to deleting a metric and introducing a new one. Accordingly, metric renaming will also be disallowed for stable metrics.
+
+## Deprecation Lifecycle
+
+Metrics can be annotated with a Kubernetes version, from which point that metric will be considered deprecated. This allows us to indicate that a metric is slated for future removal and provides the consumer a reasonable window in which they can make changes to their monitoring infrastructure which depends on this metric.
+
+While deprecation policies only actually change stability guarantees for __stable__ metrics (and not __alpha__ ones), deprecation information may however be optionally provided on alpha metrics to help component owners inform users of future intent, to help with transition plans (this change was made at the request of @dashpole, who helpfully pointed out that it would be nice to be able signal future intent even for alpha metrics).
+
+When a stable metric undergoes the deprecation process, we are signaling that the metric will eventually be deleted. The lifecyle looks roughly like this (each stage represents a Kubernetes release):
```

Comment:
Agree on roughly. In a few cases, it might don't have stage "__Hidden metric__".    For example, if a component is removed, then related metrics are moved.